### ISSUE TITLE
Cull geometry at the instance level

### ISSUE THREAD

**Issue by HoyosJuan** [ThatOpen/engine_components#248]
### Description 📝

The current ScreenCuller hide entire FragmentMeshes, not it's instances. This feature is to propose the implementation of an Fragment Instance Culler.

### Validations ✅

- [X] Read the [docs](https://docs.thatopen.com/intro).
- [X] Check that there isn't [already an issue](https://github.com/IFCjs/components/issues) that requests the same feature to avoid creating a duplicate.

**Comment by RyugaRyuzaki** [comment]
This topic is so great @HoyosJuan .I also have an idea to speed up culling. like in this code https://github.com/IFCjs/components/blob/main/src/core/ScreenCuller/index.ts#L263-L272, we use GPU to handle the culling process. The question is Do we implement it under worker and instead of just 1 we create 30-40 workers to handle it? like usBIM, I guess they used it.(as shown in the picture). So I went to a pretty old library https://github.com/stackgl/headless-gl. But I got an error when installing that I couldn't pre-build it. The idea is that for 1 model we can have 5000-10000 FragmentMeshes. Now I divide it into 1 culling worker that will process 1000 FragmentMeshes.
![image](https://github.com/IFCjs/components/assets/89787521/a6eec8ba-f8e8-4133-9c63-ebce54269031)
